resolwe_bio/tests/processes/test_expression.py

```diff
@@ -494,6 +494,197 @@ def test_feature_counts(self):
+            compression="gzip",
+        )
+        self.assertFile(
+            expression,
+            "cpm",
+            outputs / "feature_counts_out_cpm.tab.gz",
+            compression="gzip",
+        )
+        self.assertFile(
+            expression,
+            "exp",
+            outputs / "feature_counts_out_tpm.tab.gz",
+            compression="gzip",
+        )
+        self.assertFile(
+            expression,
+            "exp_set",
+            outputs / "feature_counts_out_exp_set.txt.gz",
+            compression="gzip",
+        )
+        self.assertJSON(
+            expression,
+            expression.output["exp_set_json"],
+            "",
+            outputs / "feature_counts_exp_set.json.gz",
+        )
+        self.assertFields(expression, "species", "Homo sapiens")
+        self.assertFields(expression, "build", "GRCh38_ens90")
+        self.assertFields(expression, "feature_type", "gene")
+
+        # test using BAM file containing single-end reads and a GFF input file
+        expression = self.run_process(
+            "feature_counts-beta",
+            {
+                "aligned_reads": bam_single.id,
+                "annotation": annotation_gff3.id,
+                "id_attribute": "Parent",
+            },
+        )
+        self.assertFile(
+            expression,
+            "rc",
+            outputs / "feature_counts_out_gff3_rc.tab.gz",
+            compression="gzip",
+        )
+        self.assertFile(
+            expression,
+            "exp",
+            outputs / "feature_counts_out_gff3_tpm.tab.gz",
+            compression="gzip",
+        )
+        self.assertFields(expression, "feature_type", "gene")
+
+        expression_lanes = self.run_process(
+            "feature_counts-beta",
+            {
+                "aligned_reads": single_lanes.id,
+                "annotation": annotation.id,
+            },
+        )
+        self.assertFile(
+            expression_lanes,
+            "rc",
+            outputs / "feature_counts_single_lanes_rc.tab.gz",
+            compression="gzip",
+        )
+
+        # test using UCSC-derived annotation
+        expression = self.run_process(
+            "feature_counts-beta",
+            {
+                "aligned_reads": bam_ucsc.id,
+                "annotation": annotation_ucsc.id,
+            },
+        )
+        self.assertFile(
+            expression,
+            "rc",
+            outputs / "feature_counts_out_ucsc_rc.tab.gz",
+            compression="gzip",
+        )
+        self.assertFile(
+            expression,
+            "exp",
+            outputs / "feature_counts_out_ucsc_tpm.tab.gz",
+            compression="gzip",
+        )
+
+        expression_lanes = self.run_process(
+            "feature_counts-beta",
+            {
+                "aligned_reads": paired_lanes.id,
+                "annotation": annotation.id,
+            },
+        )
+        self.assertFile(
```

Comment:
If the process modified the `feature_counts_output` the tests would have to cover that as well.